htsinfer/infer_count_motifs.py

```diff
@@ -0,0 +1,91 @@
+"""Infer motif frequency from sample data."""
+
+
+from typing import Dict
+
+
+def count_motifs(input_sequences: list, min_motif_length: int,
+                 max_motif_length: int, nucleic_acid: str) -> dict:
+    """Function that calculates the occurrence of all motifs in one or
+    multiple sequences and returns a dictionary with all motifs within
+    the specified length and their occurrence.
+
+        Args:
+            input_sequences (list): list of sequences
+            min_motif_length (int): minimal length of motif
+            max_motif_length (int): maximal length of motif
+            nucleic_acid (string): type of nucleic acid (dna, rna)
+
+        Returns:
+            sorted_dict (dict): paired motif frequency {motif: frequency }
+
+        Raises:
+            TypeError: input_sequences is not a list
+            TypeError: min_motif_length or max_motif_length is not an integer
+            ValueError: nucleic_acid is not in specified range
+            ValueError: max_motif_length is longer than the shortest list
+    """
+
+    if not isinstance(input_sequences, list):
```

Comment:
The input you'll get in this function will be passed from another function, and the type hints will make sure that we will realize if what we pass is not of the correct type. Thus, when using type hints, it is really only necessary to validate actual raw user input.

Also note that you might feel that, having done your type checks, you are going to be safe when actually using the variables, but in Python, the type of a variable can change between one value and another. Even if you're sure that you don't put any code in between your validation and the place(s) where you use a variable that may change the type, someone else _may_ do so in the future, and the validation might give a false sense of security. If you don't want to have your script just fail in a case like that, the best thing is to put your potentially failing code in a `try - except` block and specifically catch any exceptions that you can think of.